app/views/pages/developer-guide/user-uploads/user-uploads.liquid

```diff
@@ -65,22 +65,26 @@ properties:
     }
   }
 {% endgraphql %}
-<form action="{{ data.property_upload_presigned_url.upload_url }}">
+<form action="{{ data.property_upload_presigned_url.upload_url }}" enctype="multipart/form-data" method="post">
   {% for field in data.property_upload_presigned_url.upload_url_payload %}
     <input type="hidden" name="{{ field[0] }}" value='{{ field[1] }}'>
   {% endfor %}
+  <button type="submit">Upload</button>
 </form>
 ```
 {% endraw %}
 
 3. S3: Return URL to file
-  S3 will return status 204 and an XML response with the URL to the file on S3, if everything went well. Otherwise it will return status 4xx and the reason for the error in XML format.
+  After submitting such form, you will be redirected to an XML response (204 status code) with the URL to the file on the storage servers. Otherwise it will return status 4xx and the reason for the error in XML format.
 
-4. Browser: POST URL to application
-  Until now, platformOS doesn't know anything about the file that the user has been uploading. This step is meant to change that. Using any method you wish, you need to send the URL to the uploaded file to the application server. Because `file` is just an URL (string) property in a record, you update that property just like any other.
+  Please note that for optimal user experience you would likely want to to this using an `fetch` request from JavaScript, so that the whole upload process stays invisible to end user.
```

Comment:
```suggestion
  Please note that, for an optimal user experience, you likely want to use a `fetch` request provided by JavaScript, so that the entire upload process stays invisible to the end user.
```